backend/python/app/services/implementations/story_service.py

```diff
@@ -240,3 +243,22 @@ def get_story_translations_available_for_review(self, language, level):
         except Exception as error:
             self.logger.error(str(error))
             raise error
+
+    def get_translation_progress(self, story_id, translation_id):
+        try:
+            num_translated_lines = (
+                db.session.query(StoryTranslationContent)
+                .filter(StoryTranslationContent.story_translation_id == translation_id)
+                .count()
+            )
+            num_story_lines = (
+                db.session.query(StoryContent)
+                .filter(StoryContent.story_id == story_id)
+                .count()
+            )
+            percentage_complete = round(num_translated_lines / num_story_lines, 2) * 100
```

Comment:
ahh sorry i had meant to add in a check for non-empty strings as well like this, but i accidentally deleted it when i added in the count():
```
num_translated_lines = (
                db.session.query(StoryTranslationContent)
                .filter(
                    StoryTranslationContent.story_translation_id == translation_id,
                    StoryTranslationContent.translation_content != "",
                )
                .count()
            )
```
would that work to return the correct percentage?

i can definitely move it into `get_story_translation`--would the translationContent that gets returned in `StoryTranslationResponseDTO` only include the non-empty string values? also how often do we want the progress bar to be updated, and do we need to call get_story_translation multiple times on the translation page as well?

Reply:
Heads up that
> alternative idea: modify StoryTranslationResponseDTO to return translationContent (@e-wai already has this in her WIP pr #39 )

is currently not working

Reply:
>[content in dto] is currently not working

@e-wai  We can extend this conversation on #39 but my guess is that StoryTranslation model needs something like the `content` SQLAlchemy attribute in the Story model

> would the translationContent that gets returned in StoryTranslationResponseDTO only include the non-empty string values?

If we follow the above suggestion, the query would return all content lines regardless of actual string content. 

> also how often do we want the progress bar to be updated, and do we need to call get_story_translation multiple times on the translation page as well?

We're taking a [direct to database approach (stackoverflow ans)](https://stackoverflow.com/questions/44435034/auto-save-server-architecture) for now to auto save translation changes. So we will certainly be making alot of update mutations - I think we should write these updates to return the storytranslation + it's content, and therefore we can return the `percentageComplete` too

Reply:
Do we want to store `percentageComplete` as a column on StoryTranslation?
- on update (mutation) -> recalculate. However, if we were to add as a column, we would have be editing two tables (is this expensive?)
- on query for story translation -> if it's stored as a column, this isn't a problem and can just a direct read

Mini optimization frontend -> having logic to check existing string value, "" vs. "abc". If it's the latter, don't even bother with the recalculate portion of the call since any changes won't affect progress.
- However, there's an edge case where if the user is going from "blah" to "", should we be toggling progress back?